tests/unit/test-class-wc-payment-gateway-wcpay.php

```diff
@@ -3969,6 +3969,50 @@ public function is_proper_intent_used_with_order_returns_false() {
 		$this->assertFalse( $this->card_gateway->is_proper_intent_used_with_order( WC_Helper_Order::create_order(), 'wrong_intent_id' ) );
 	}
 
+
+	public function test_get_recommended_payment_method() {
+		$this->mock_wcpay_account
+			->expects( $this->once() )
+			->method( 'get_recommended_payment_methods' )
+			->with( 'US' );
+		$this->card_gateway->get_recommended_payment_methods( 'US' );
+	}
+
+
+	public function get_recommended_payment_method_no_country_code_provider() {
+		return [
+			'provider connected'     => [ true, 'test' ],
+			'provider not connected' => [ false, 'US' ],
+		];
+	}
+
+	/**
+	 * @dataProvider get_recommended_payment_method_no_country_code_provider
+	 */
+	public function test_get_recommended_payment_method_no_country_code_provided( $is_provider_connected, $country_code ) {
+		$this->mock_wcpay_account
+			->expects( $this->once() )
+			->method( 'is_provider_connected' )
+			->willReturn( $is_provider_connected );
+
+		$this->mock_wcpay_account
+			->expects( $this->any() )
+			->method( 'is_stripe_connected' )
+			->willReturn( true );
+
+		$this->mock_wcpay_account
+			->expects( $this->any() )
+			->method( 'get_account_country' )
+			->willReturn( 'test' );
```

Comment:
Shouldn't this be `->willReturn( $country_code );`?

Reply:
the code will be called only for one of the tests from the provider. anyway changed it [3af3993](https://github.com/Automattic/woocommerce-payments/pull/9825/commits/3af399332f1bc9159887a0f09db97ea47fa75fcb)

Reply:
> the code will be called only for one of the tests from the provider

I don't follow. Could you elaborate, please?